src/util/kaldi-table-inl.h

```diff
@@ -302,6 +325,24 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
       if (!key_.empty() && !data_rxfilename_.empty()) {
         // Got a valid line.
         state_ = kHaveScpLine;
+        // Here we "peel off" the range_ specifer (if any, e.g. [1:2,2:10])
+        // from data_rxfilename_ (e.g. 1.ark:100).
+        if (data_rxfilename_.find_last_of("]") == data_rxfilename_.length() - 1) {
```

Comment:
OK then you can do

data_rxfilename_[data_rxfilename_.size()-1]



On Wed, Mar 23, 2016 at 1:41 AM, Xiaohui Zhang <notifications@github.com>
wrote:

> In src/util/kaldi-table-inl.h
> <https://github.com/kaldi-asr/kaldi/pull/611#discussion_r57112249>:
>
> > @@ -302,6 +325,24 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
> >        if (!key_.empty() && !data_rxfilename_.empty()) {
> >          // Got a valid line.
> >          state_ = kHaveScpLine;
> > +        // Here we "peel off" the range_ specifer (if any, e.g. [1:2,2:10])
> > +        // from data_rxfilename_ (e.g. 1.ark:100).
> > +        if (data_rxfilename_.find_last_of("]") == data_rxfilename_.length() - 1) {
>
> hmm I tried this and it needs c++11..
> … <#160354721791884107_>
> On Mar 23, 2016 1:37 AM, "Daniel Povey" <notifications@github.com> wrote:
> In src/util/kaldi-table-inl.h <#611 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/611#discussion_r57112080>>: > @@
> -302,6 +325,24 @@ template<class Holder> class
> SequentialTableReaderScriptImpl: > if (!key_.empty() &&
> !data_rxfilename_.empty()) { > // Got a valid line. > state_ =
> kHaveScpLine; > + // Here we "peel off" the range_ specifer (if any, e.g.
> [1:2,2:10]) > + // from data_rxfilename_ (e.g. 1.ark:100). > + if
> (data_rxfilename_.find_last_of("]") == data_rxfilename_.length() - 1) { if
> (data_rxfilename_.back() == ']') is easier — You are receiving this because
> you authored the thread. Reply to this email directly or view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/611/files/75a3ad3d057e3aec153697c44dc23e070adf9076#r57112080>
>
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/611/files/75a3ad3d057e3aec153697c44dc23e070adf9076#r57112249>
>